Base: Add many man pages

demostanis opened the pull request:
Still a draft since I want to add more man pages and make the synopsis of the current ones better.

LucasChollet commented:
I would advise you to stop adding contents to this PR before it becomes a pain for maintainers to review.

bgianfo commented on Base/usr/share/man/man1/nl.md:
````diff
@@ -1,11 +1,11 @@
 ## Synopsis
 
 ```sh
-$ nl [--body-numbering style] [--increment number] [--separator string] [--startnum number] [--width number] [file...]
````
Why remove the list of options?

demostanis replied:
At first I wanted to have it look like other man pages.
But after thinking a bit, IMO having options in the synopsis only makes sense to inform about potential clashes between options (i.e. a `|`, meaning two options can't be used altogether). Otherwise it doesn't have any benefit over the `Options` section.

demostanis replied:
(thus I only removed the list of options from the synopsis in a few man pages)

bgianfo commented on Base/usr/share/man/man1/grep.md:
```diff
@@ -1,31 +1,56 @@
+## See also
 
-* `file`: File(s) to process
+* [POSIX Regular expressions](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html)
```
```suggestion
* [POSIX Regular Expressions](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html)
```

bgianfo commented on Base/usr/share/man/man1/grep.md:
```diff
@@ -1,31 +1,56 @@
+* `--binary-mode`: Action to take for binary files: binary (default), text, skip
+* `-a`, `--text`: Treat binary files as text (same as `--binary-mode text`)
+* `-I`: Ignore binary files (same as `--binary-mode skip`)
+* `--color WHEN`: When to use colored output for the matching text: auto (default), never, always
```
If you want to fix the format, why not fix the `ArgParser` generator so it generates them in the desired format?

demostanis replied:
Sure thing, but that's out of scope for this PR

demostanis commented:
changes since last push:
improved passwd(1)'s synopsis

demostanis commented:
changes since last push (i promise those are the last):
- update tr(1)'s synopsis
- remove group(4) reference in passwd(4) (it doesn't exist)
- correct the reference to proc(7) in lsof(1)

martinfalisse commented on Base/usr/share/man/man4/passwd.md:
```diff
@@ -0,0 +1,29 @@
+- The second field is the encrypted password. It is always a `!`, to indicate that the real password is in `/etc/shadow`.
+- The third field is the user ID.
+- The fourth field is the main group ID.
+- The fifth field is the GECOS username, a longer string to indicate general information about the user.
```
.. username; a longer string (use semicolon)

demostanis replied:
what about a colon? the two people i've asked say it sounds better with a colon

martinfalisse commented on Base/usr/share/man/man4/passwd.md:
```diff
@@ -0,0 +1,29 @@
+
+## Examples
+
+* `root:!:0:0:root:/root:/bin/Shell`: The default fields for the root user. Its home directory is `/root`, and its uid and gid are both 0.
```
Capitalize UID and GID

martinfalisse commented on Base/usr/share/man/man4/shadow.md:
```diff
@@ -0,0 +1,26 @@
+- The first field is the username.
+- The second field is the SHA256-encrypted password. If it starts with a `!`, the account is locked.
+- The third field is the date at which the password was last changed, as a timestamp. `useradd` currently hardcodes this to 18727.
+The rest of the fields don't have any effects.
```
"are unused", maybe

martinfalisse commented on Base/usr/share/man/man4/shadow.md:
```diff
@@ -0,0 +1,26 @@
+## Name
+
+passwd - Encrypted passwords
```
I personally would have liked something more descriptive here, like "System's users and password information"

demostanis replied:
right, and i also just realized that's the wrong name lol

martinfalisse commented on Base/usr/share/man/man1/nl.md:
```diff
@@ -1,11 +1,11 @@
 ## Name
 
-nl
+nl - number lines of files
```
Think "number" should be capitalized.

martinfalisse commented on Base/usr/share/man/man1/nl.md:
```diff
@@ -18,6 +18,21 @@ $ nl [--body-numbering style] [--increment number] [--separator string] [--start
-<!-- Auto-generated through ArgsParser -->
+## Examples
+
+Number lines in README.md:
```
Maybe "Add numbering to lines in .. ".

martinfalisse commented on Base/usr/share/man/man1/tr.md:
````diff
@@ -1,22 +1,46 @@
+
+Remove "a"s from standard input:
+```sh
+$ echo "Weaaaaaaaaaaaaall Hello Friends" | tr -d a
````
Is there supposed to be the resulting echo below here? "Well Hello Friends" ?

martinfalisse commented on Base/usr/share/man/man1/lsof.md:
```diff
@@ -1,26 +1,42 @@
+* `-p pid`: Filter by PID
+* `-d fd`: Filter by file descriptor
+* `-u login/UID`: Filter by login/UID
+* `-g PGID`: Filter by process group ID
```
I feel like if PGID is capitalized here, then PID and FD should be capitalized above

demostanis replied:
I left them lowercase in utmpupdate. Should I make them uppercase there too? I personally prefer lowercase.

demostanis replied:
Also `username` sounds better than `login`,  I changed that.

martinfalisse commented on Base/usr/share/man/man1/lsof.md:
```diff
@@ -1,26 +1,42 @@
 ## Description
 
-List open files of a processes. This can mean actual files in the file system, sockets, pipes, etc.
+`lsof` lists opened files by certain processes, such as pipes, sockets, regular files...
```
First I would say lsof lists all opened files. Then I would say there is the possibility to add arguments to filter these files by certain processes.

martinfalisse commented on Base/usr/share/man/man1/ntpquery.md:
```diff
@@ -1,21 +1,28 @@
+## Description
+
+`ntpquery` contacts a NTP server (by default time.google.com), and optionally sets or
+adjusts (i.e. slow down or speed up) the system time, as long as the command is run as root.
```
Perhaps the mention that this needs to be run as root is unnecessary. At least, I don't see this same type of comment having been added to the other manpages that required a certain privilege.

demostanis replied:
`groupadd(8)` does mention that it needs to be run as root.

demostanis commented:
CI flake
